### PR TITLE
e2e: Remove the cache with "podman unshare rm" when a rootless user

### PR DESCRIPTION
**Pull request by sstosh**
If rootless, os.RemoveAll() is failed due to permission denied.
Therefore, we use "podman unshare rm" instead of os.RemoveAll().

Signed-off-by: Toshiki Sonoda <sonoda.toshiki@fujitsu.com>

---

Related to #17089

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: test/e2e/common_test.go [hunk 1]
```diff
@@ -556,8 +556,16 @@ func (p *PodmanTestIntegration) Cleanup() {
...
+	if isRootless() {
+		// If rootless, os.RemoveAll() is failed due to permission denied
+		cmd := exec.Command(p.PodmanBinary, "unshare", "rm", "-rf", p.TempDir)
+		if err := cmd.Run(); err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+		}
+	} else {
+		if err := os.RemoveAll(p.TempDir); err != nil {
+			fmt.Printf("%q\n", err)
+		}
```
Comment (Luap99) [inline test/e2e/common_test.go:568]: please put this is a function that accepts the path as arg so we do not duplicate the same lines twice

**Comment by rhatdan** [comment]
@sstosh Looks like this needs a rebase.

File: test/e2e/common_test.go [hunk 2]
```diff
@@ -556,8 +556,16 @@ func (p *PodmanTestIntegration) Cleanup() {
 
 	p.StopRemoteService()
 	// Nuke tempdir
-	if err := os.RemoveAll(p.TempDir); err != nil {
-		fmt.Printf("%q\n", err)
+	if isRootless() {
+		// If rootless, os.RemoveAll() is failed due to permission denied
+		cmd := exec.Command(p.PodmanBinary, "unshare", "rm", "-rf", p.TempDir)
```
Comment (Luap99) [inline test/e2e/common_test.go:561]: please connect both stdout/err so we can see the error message in the logs, i.e. add
```
cmd.Stdout = os.Stdout
cmd.Stderr = os.Stderr
```

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**